src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -534,6 +534,30 @@ private boolean hasEmptyLineBefore(DetailAST token) {
     }
 
     /**
+     * Check if token is preceded by javadoc comment
+     * @param token token for check.
+     * @return true, if token is predicted by javadoc comment.
+     */
+    private boolean isPrecededByJavadoc(DetailAST token) {
+        final int lineNo = token.getLineNo();
+        boolean result = true;
+        //  [lineNo - 2] is the number of the previous line because the numbering starts
+        //  from zero.
+        int lineIndex = lineNo - 2;
+        final String[] lines = getLines();
+        String lineBefore = lines[lineIndex].trim();
+        while (!lineBefore.startsWith("/**")) {
+            if (lineIndex == 0 || !lineBefore.isEmpty() && lineBefore.charAt(0) != '*') {
```

Comment:
Taken from: http://docs.oracle.com/javase/7/docs/technotes/tools/solaris/javadoc.html#packagecomment
````
Note that while the comment separators /** and /* must be present, the leading asterisks
on the intermediate lines can be omitted.
````
Taken from: http://docs.oracle.com/javase/7/docs/technotes/tools/solaris/javadoc.html#leadingasterisks
````
Leading asterisks - When javadoc parses a doc comment, leading asterisk (*) characters on
each line are discarded; blanks and tabs preceding the initial asterisk (*) characters are also
discarded. Starting with 1.4, if you omit the leading asterisk on a line, the leading white
space is no longer removed. This enables you to paste code examples directly into a doc
comment inside a <PRE> tag, and its indentation will be honored. Spaces are generally
interpreted by browsers more uniformly than tabs. Indentation is relative to the left
margin (rather than the separator /** or <PRE> tag).
````

We should accept Javadocs with no leading asterisks. If they omit the leading asterisk, your current code will fail thinking it is not part of the Javadoc as your `lineBefore.charAt(0) != '*'` is looking for the leading asterisk.
Since we are looking above the `package` it can't be another node as there are no **valid** nodes that go above the package. We can only run into Javadocs or Comments.

I think we should change the leading asterisk check to a leading comment check (`//` or `/*`).
We should also have a test for this.

@romani Do you agree?